src/libraries/System.Text.Encodings.Web/src/System/Text/Encodings/Web/DefaultJavaScriptEncoder.cs

```diff
@@ -87,21 +88,30 @@ public override unsafe int FindFirstCharacterToEncodeUtf8(ReadOnlySpan<byte> utf
                 int idx = 0;
 
 #if NETCOREAPP
-                if (Sse2.IsSupported)
+                if (Sse2.IsSupported || AdvSimd.Arm64.IsSupported)
                 {
                     sbyte* startingAddress = (sbyte*)ptr;
                     while (utf8Text.Length - 16 >= idx)
                     {
                         Debug.Assert(startingAddress >= ptr && startingAddress <= (ptr + utf8Text.Length - 16));
 
-                        // Load the next 16 bytes.
-                        Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                        bool containsNonAsciiBytes;
 
-                        // Check for ASCII text. Any byte that's not in the ASCII range will already be negative when
-                        // casted to signed byte.
-                        int index = Sse2.MoveMask(sourceValue);
+                        // Load the next 16 bytes, and check for ASCII text.
+                        // Any byte that's not in the ASCII range will already be negative when casted to signed byte.
+                        if (Sse2.IsSupported)
+                        {
+                            Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
+                            containsNonAsciiBytes = Sse2.MoveMask(sourceValue) != 0;
+                        }
+                        else
+                        {
+                            Debug.Assert(AdvSimd.Arm64.IsSupported);
+                            Vector128<sbyte> sourceValue = AdvSimd.LoadVector128(startingAddress);
+                            containsNonAsciiBytes = AdvSimd.Arm64.MinAcross(sourceValue).ToScalar() < 0;
+                        }
```

Comment:
The original code was not wrapped by an `IsSupported` condition, meaning it should be considered the "default" option. If you invert the if/else, it would allow you to remove the `Debug.Assert` check.

So consider the following suggestion:

```suggestion
                        if (AdvSimd.Arm64.IsSupported)
                        {
                            Vector128<sbyte> sourceValue = AdvSimd.LoadVector128(startingAddress);
                            containsNonAsciiBytes = AdvSimd.Arm64.MinAcross(sourceValue).ToScalar() < 0;
                        }
                        else
                        {
                            Vector128<sbyte> sourceValue = Sse2.LoadVector128(startingAddress);
                            containsNonAsciiBytes = Sse2.MoveMask(sourceValue) != 0;
                        }
```

Reply:
I'm intentionally testing for the Sse2 predicate here, since the AdvSimd equivalent is a shimmed `false` constant in targets < net5.0 and results in compiler warnings. The assertion is redundant, but is meant to guard against bugs during potential future additions of other architectures.

Reply:
> You could also declare sourceValue outside of the if/else, since it's used in both.

It would make sense to do this if the variable escaped the scope of the particular branching. Since it doesn't, I would rather keep it like this to avoid potential bugs.